controllers/runner_controller.go

```diff
@@ -552,6 +637,10 @@ func newRunnerPod(runnerName string, template corev1.Pod, runnerSpec v1alpha1.Ru
 			Name:  EnvVarEphemeral,
 			Value: fmt.Sprintf("%v", ephemeral),
 		},
+		{
+			Name:  "RUNNER_CUSTOM_RBAC",
```

Comment:
I'd prefer naming this envvar more around for what it works, like:

```suggestion
			Name:  "RUNNER_STATUS_UPDATE_HOOK",
```

The use of a custom RBAC can conceptually be optional if one prefers using a shared rolebinding and a serviceaccount that have broader permissions

Reply:
I'm still generally afraid of using `RUNNER_` variables for ARC specific things. GitHub uses them and we are only asking for a future collision by continuing this practice. We really should switch to prefixing them with `ARC_`.

We also need a way to tell variables apart that are meant to be passed down to the runner, and those that should not be passed down. (You surely remember the automatic, prefix-based unsetting idea for the entrypoint script.) Not sure how we could do that in a sustainable way.

Reply:
Updated to RUNNER_STATUS_UPDATE_HOOK for now as prefixing env variables may be of a larger scope for this PR as most likely we'll want consistency in all env vars

Reply:
> GitHub uses them

I thought they tend to use `ACTIONS_RUNNER_` prefix these days. Do you have any references on the use of `RUNNER_`-prefixed envvars by GitHub?(Probably I lost my memory? 🤔

Reply:
Anyway, it would be nicer if we could migrate to a more unique prefix in the future, like `ARC_`.